sympy/core/local_random.py

```diff
@@ -0,0 +1,20 @@
+"""
+This module is a copy of python's random module but using
+a Random instance for all supported methods. It is needed
+for replicating results when using seeds.
+"""
+
+import random as std_random
+import numpy
+import sys
+
+_inst = std_random.Random()
+_thismodule = sys.modules[__name__]
+
+for method_name in std_random.__all__:
+    in_inst = hasattr(_inst, method_name)
+    parent = _inst if in_inst else std_random
+    method = getattr(parent, method_name)
+    setattr(_thismodule, method_name, method)
+
+numpy_random = numpy.random.RandomState
```

Comment:
A newline  is needed here to pass the code quality checks.

Reply:
You mean an empty line at the end (21) or another one in line 19?

Reply:
Looping through `std_random.__all__` is a nice idea but I don't think it's what we want to do here.

This module should explicitly import the functions that we want so that they are clearly listed here.

Also we shouldn't import numpy except perhaps with `import_module` because it isn't a hard dependency for SymPy.

Reply:
Fixed